sdk/appcenter-data/src/main/java/com/microsoft/appcenter/data/Data.java

```diff
@@ -561,6 +576,7 @@ public boolean needsRemoteOperation(DocumentWrapper<Void> cachedDocument) {
 
             @Override
             public void callCosmosDb(TokenResult tokenResult, DefaultAppCenterFuture<DocumentWrapper<Void>> result) {
+                mLocalDocumentStorage.deleteOffline(Utils.getTableName(tokenResult.getPartition()), tokenResult.getPartition(), documentId, writeOptions);
```

Comment:
Accessing storage from main thread is not allowed in strict mode. (HTTP callbacks are executed in main/ui thread).

Reply:
hmm I got the point, but the tricky part for this is we need to know what is the table name to write.... And the way we get the table name is to parse the token result from the token-exchange service.
And we can use the cache token for sure ":), but.... the first time will not have cache... So not sure how to refactory it...

Reply:
changed to async, not sure if correct :D